### PR TITLE
F1, X1, USB Cancels

### PR DESCRIPTION
**Pull request by harryhaaren**
General development, F1, X1 and usb cancels fixed.

### CODE REVIEW

File: ctlra/devices/ni_kontrol_x1_mk2.c [hunk 1]
```diff
@@ -456,10 +466,91 @@ ni_kontrol_x1_mk2_light_flush(struct ctlra_dev_t *base, uint32_t force)
...
+
+	dev->lights_81[0] = 0x81;
+	ret = ctlra_dev_impl_usb_interrupt_write(base,
 					   USB_HANDLE_IDX,
 					   USB_ENDPOINT_WRITE,
-					   data, LIGHTS_SIZE+1);
+					   dev->lights_81,
+					   IFACE_Ox81_TOTAL);
+	if(ret != IFACE_Ox81_TOTAL) {
+		//printf("%s: %d ret = %d, size =\n", __func__, dev->lights_interface, ret);
```
Comment (foolswood) [inline ctlra/devices/ni_kontrol_x1_mk2.c]: Unfinished?

File: ctlra/devices/ni_kontrol_x1_mk2.c [hunk 2]
```diff
@@ -468,10 +559,14 @@ ni_kontrol_x1_mk2_disconnect(struct ctlra_dev_t *base)
...
-	memset(&dev->lights[1], 0, NI_KONTROL_X1_MK2_LED_COUNT);
-	dev->lights[0] = 0x80;
-	if(!base->banished)
+	memset(dev->lights, 0, NI_KONTROL_X1_MK2_LED_COUNT);
+	memset(dev->lights_81, 0, sizeof(dev->lights_81));
+	if(!base->banished) {
 		ni_kontrol_x1_mk2_light_flush(base, 1);
+	}
+	if(!base->banished) {
+		ni_kontrol_x1_mk2_light_flush(base, 1);
```
Comment (foolswood) [inline ctlra/devices/ni_kontrol_x1_mk2.c]: Double flush?

File: ctlra/devices/ni_kontrol_x1_mk2.c [hunk 3]
```diff
@@ -510,9 +605,23 @@ ctlra_ni_kontrol_x1_mk2_connect(ctlra_event_func event_func,
 	dev->base.light_set = ni_kontrol_x1_mk2_light_set;
 	dev->base.light_flush = ni_kontrol_x1_mk2_light_flush;
 	dev->base.usb_read_cb = ni_kontrol_x1_mk2_usb_read_cb;
+	dev->base.feedback_digits = ni_kontrol_x1_mk2_feedback_digits;
 
 	dev->base.event_func = event_func;
 	dev->base.event_func_userdata = userdata;
+#if 1
```
Comment (foolswood) [inline ctlra/devices/ni_kontrol_x1_mk2.c]: Looks like you don't need the preprocessor foo :)

File: ctlra/devices/ni_kontrol_x1_mk2.c (hunk 1 above)
Comment (harryhaaren) [reply ctlra/devices/ni_kontrol_x1_mk2.c]: Fixed

File: ctlra/devices/ni_kontrol_x1_mk2.c (hunk 2 above)
Comment (harryhaaren) [reply ctlra/devices/ni_kontrol_x1_mk2.c]: Fixed

File: ctlra/devices/ni_kontrol_x1_mk2.c (hunk 3 above)
Comment (harryhaaren) [reply ctlra/devices/ni_kontrol_x1_mk2.c]: Fixed, thanks!

**Pull request merged**